### PR TITLE
docs: Document how to connect using OIDC

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: docs/modules/trino/pages/usage-guide/connect_to_trino.adoc [hunk 1]
```diff
@@ -78,3 +88,28 @@ connection = get_connection()
...
+
+In case you are using OpenID connect use the following connection definition:
+
+[source,python]
+----
+    connection = trino.dbapi.connect(
+        host="87.106.119.4", # You need to use an an externally reachable address, so that your browser can access it!
+        port=8443,
+        http_scheme="https",
+        verify=False, # For best security you can also provide a path to the trino root ca: "/stackable/secrets/trino-ca-cert/ca.crt",
```
Comment (NickLarsenNZ) [inline docs/modules/trino/pages/usage-guide/connect_to_trino.adoc:100]: We should probably give the secure example, and then offer False in the comment.

File: docs/modules/trino/pages/usage-guide/connect_to_trino.adoc [hunk 2]
```diff
@@ -47,6 +53,8 @@ image::connect-with-dbeaver-3.png[]
 
 As the last step you can click on _Finish_ and start using the Trino connection.
 
+TIP: In case you are using OpenID connect, set the `externalAuthentication` property to `true` and don't provide and username or password. A browser window will be opened, which might require you to log in.
```
Comment (NickLarsenNZ) [inline docs/modules/trino/pages/usage-guide/connect_to_trino.adoc:56]: Did this recently change? I recall the `--user` flag being required even with external auth (else $USER is used), so I assume that got fixed. If so, can we link to the issue for posterity?

Comment (sbernauer) [reply docs/modules/trino/pages/usage-guide/connect_to_trino.adoc:56]: Good catch, this is still an issue! Should be fixed in 8027f64

File: docs/modules/trino/pages/usage-guide/connect_to_trino.adoc (hunk 1 above)
Comment (sbernauer) [reply docs/modules/trino/pages/usage-guide/connect_to_trino.adoc:100]: First off: I think the important part is documenting the secure path.

I personally would prefer to ship an working example, especially as sadly my feeling is many of the users are not knowledge enough to get the ca.crt, we have tracking issues to improve that situation, e.g. https://github.com/stackabletech/secret-operator/issues/410

Comment (NickLarsenNZ) [reply docs/modules/trino/pages/usage-guide/connect_to_trino.adoc:100]: I get that the examples should be working, but can we not have both secure and working? (With autoTls)

Feel free to say no, and mark as resolved, I just wanted to discuss here so I can understand why not.

Comment (sbernauer) [reply docs/modules/trino/pages/usage-guide/connect_to_trino.adoc:100]: >  but can we not have both secure and working? (With autoTls)

I execute the script on my local machine (or JupyterHub). So I first need to follow the workaround from https://github.com/stackabletech/secret-operator/issues/410, extract the ca.crt and place it on my local machine.
I can do that easily but I fear many users will have problems with that/are not able to do that.

So I don's see any config that work on all the developer machines out there :see_no_evil: 
But I can tell you that the python client is *very* good in annoying you with many many warnings :D

Comment (NickLarsenNZ) [reply docs/modules/trino/pages/usage-guide/connect_to_trino.adoc:100]: Oh I see. It's just for local development from a python script. I thought it was configuring a deployed thing which should definitely use the secure example even if it is "harder".

So I'm fine with this.

Comment (sbernauer) [reply docs/modules/trino/pages/usage-guide/connect_to_trino.adoc:100]: Oh yeah, this is *not* a k8s resource where we could point to a SecretClass, but a random Python script instead :+1:

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**